tests/unit/helpers.py

```diff
@@ -18,6 +18,38 @@
 import pytest
 
 
+def add_header_assertion_to_kwargs(kwargs):
```

Comment:
My initial reaction to this is duplicating too much logic from the actual implementation and encouraging more "change detector tests". The need for it probably indicates we've been too strict on our header assertions to begin with.

Note to self: any additional thoughts after seeing the rest of the test updates?

Reply:
This is a clever solution. I think my initial reaction is still correct, but it's not worth it to rip out our (probably too low level) `api_request` call assertions for something less specific.

If we were writing these tests for the first time, I think separate assertions for specific headers would be desired.

Reply:
Not only specific header assertions, but a separate assertion for path, a separate assertion for body, etc.

Reply:
Here's another idea, perhaps benefiting from a weekend's distance. :)

Abstract the addition of the `X-Server-Timeout` header into a function.  Add an autouse test fixture that replaces that function with a noop except on the tests that test addition of the header, by adding a special marker to those tests and and checking for the marker in the autouse fixture.

(https://stackoverflow.com/questions/38748257/disable-autouse-fixtures-on-specific-pytest-marks)

Reply:
@tswast shall I? :)

Reply:
That does sound preferable if it means we can remove the logic that checks for timeout in our tests.